### PR TITLE
Home screen

### PR DESCRIPTION
**Pull request by LebkuchenCJ**
## Deployment:
(Create a new user, sign and see the result)
https://pryces-home-screen-bgiiwuimeik.herokuapp.com/

## Comment: 
 - I adapted my empty home screening.
 - As this PR is created of an active branch, please ignore all files changed except of HOME.js, LIST.js and EMPTYLISTSCREEN.js

image

### CODE REVIEW

File: client/src/components/EmptyListScreen.js [hunk 1]
```diff
@@ -1,19 +1,23 @@
...
         <img src={logoSrc} alt="Logo" />
-        <h2>Add your products and start saving</h2>
+        <h2>{text}</h2>
       </div>
     </Container>
   );
 }
 
 export default EmptyListScreen;
+EmptyListScreen.propTypes = {
```
Comment (jamarob) [inline client/src/components/EmptyListScreen.js:18]: I like to keep PropTypes near the start of the component function where they are destructured for quick reference what is what.

File: client/src/components/EmptyListScreen.js [hunk 2]
```diff
@@ -1,19 +1,23 @@
 import styled from "@emotion/styled";
 import React from "react";
 import logoSrc from "../assets/logo.svg";
+import PropTypes from "prop-types";
 
-function EmptyListScreen() {
+function EmptyListScreen({ text }) {
   return (
     <Container>
       <div>
```
Comment (jamarob) [inline client/src/components/EmptyListScreen.js:9]: Why the inner div?

File: client/src/components/EmptyListScreen.js [hunk 3]
```diff
@@ -1,19 +1,23 @@
...
         <img src={logoSrc} alt="Logo" />
-        <h2>Add your products and start saving</h2>
+        <h2>{text}</h2>
       </div>
     </Container>
   );
 }
 
 export default EmptyListScreen;
```
Comment (jamarob) [inline client/src/components/EmptyListScreen.js:17]: Exporting the function where it is defined is prefered imho.

File: client/src/pages/Home.js [hunk 4]
```diff
@@ -8,46 +8,61 @@ import Header from "../components/Header";
 import FloatingActionButton from "../components/FloatingActionButton";
 import ListCreationContainer from "../components/ListCreationContainer";
 import { deleteList } from "../api/list";
+import EmptyListScreen from "../components/EmptyListScreen";
 
 function Home() {
   const [name, setName] = useState("");
   const [inputfield, setInputfield] = useState(false);
```
Comment (jamarob) [inline client/src/pages/Home.js]: Maybe camel case like `inputField`?

File: client/src/pages/Home.js [hunk 5]
```diff
@@ -8,46 +8,61 @@ import Header from "../components/Header";
...
+  const { userId } = sessionStorage;
+  const [userLists, setUserLists] = useState(null);
   const { data: lists, loading, error, refetch } = useAsync(fetchLists);
 
+  useEffect(() => {
+    function handleLists() {
```
Comment (jamarob) [inline client/src/pages/Home.js:21]: Why the wrapper, seems unnecessary?

File: client/src/pages/Home.js [hunk 6]
```diff
@@ -8,46 +8,61 @@ import Header from "../components/Header";
...
+  }, [lists, userId]);
+
   async function handleSubmit(event) {
     event.preventDefault();
     const currentDate = new Date();
     const date = currentDate.toLocaleDateString();
-    const data = { name, creationDate: date, products: [] };
+    const data = { name, userId, creationDate: date, products: [] };
     await postList(data);
     await refetch();
```
Comment (jamarob) [inline client/src/pages/Home.js:34]: Post and refetch seems like an extra roundtrip to the server, is this really necessary to keep your data in sync?

File: client/src/pages/Home.js [hunk 7]
```diff
@@ -8,46 +8,61 @@ import Header from "../components/Header";
...
-            <ListItem
-              key={list._id}
-              list={list}
-              href={`/home/${list._id}`}
-              onDeleteConfirm={() => handleDelete(list._id)}
-            />
-          ))}
-        </List>
-        <FloatingActionButton displayForm={() => setInputfield(!inputfield)} />
+        {userLists?.length < 1 && (
```
Comment (jamarob) [inline client/src/pages/Home.js:47]: userLists is initialized to null, which is falsy so simply do `usersList && ...` Think about turning this and the next check into a ternary.

File: client/src/pages/Home.js [hunk 8]
```diff
@@ -8,46 +8,61 @@ import Header from "../components/Header";
...
+          <EmptyListScreen text="Create a list and your groceries" />
+        )}
+        {userLists?.length > 0 && (
```
Comment (jamarob) [inline client/src/pages/Home.js:50]: Initializing `userLists` to an empty array will save you the optional chaining. Array length greater 0 are truthy, simplify to just `userLists.length && ..`. See comment above.

File: client/src/pages/Home.js [hunk 9]
```diff
@@ -8,46 +8,61 @@ import Header from "../components/Header";
...
+          <>
+            <List>
+              {error && <div>Could not get data. Please cry.</div>}
```
Comment (jamarob) [inline client/src/pages/Home.js]: Semantic tags ;)

File: client/src/pages/Home.js [hunk 10]
```diff
@@ -8,46 +8,61 @@ import Header from "../components/Header";
...
+                  list={list}
+                  href={`/home/${list._id}`}
+                  onDeleteConfirm={() => handleDelete(list._id)}
+                />
+              ))}
+            </List>
+          </>
+        )}
         {inputfield && (
           <ListCreationContainer
```
Comment (jamarob) [inline client/src/pages/Home.js:67]: ListCreationContainer sounds like a form.

File: client/src/pages/List.js [hunk 11]
```diff
@@ -87,7 +87,9 @@ function List({ onGroceryListChange }) {
           <Header title={list.name} showBackButton></Header>
 
           <Container>
-            {list.products?.length < 1 && <EmptyListScreen />}
+            {list.products?.length < 1 && (
```
Comment (jamarob) [inline client/src/pages/List.js:92]: Simplyfiy your conditionals and maybe turn this into a ternary, as they are complements of each other.

**Review by jamarob — CHANGES_REQUESTED** [review]
Hi Jonas,
left my 2 cents, mostly on naming and some possible simplifications. Please use semantic tags where appropriate. 

You define a lot of states in your components, maybe hide some of the complexity in a custom hook? 

https://github.com/JonasImm/pryces/blob/44a334007a4a6db13c9f094521de8970a12070d0/client/src/pages/List.js#L39

This **needs** to be changed, don't generate ids with `Math.random()`!! Returning the same number on consecutive calls is _perfectly fine_ for a random generator. Use a package like uuid instead.

File: client/src/components/EmptyListScreen.js (hunk 2 above)
Comment (LebkuchenCJ) [reply client/src/components/EmptyListScreen.js:9]: For styling purpose

File: client/src/components/EmptyListScreen.js (hunk 3 above)
Comment (LebkuchenCJ) [reply client/src/components/EmptyListScreen.js:17]: This is the convention we learned and as this is implemented in all my projects for now. Will keep it mind 👍

File: client/src/pages/Home.js (hunk 5 above)
Comment (LebkuchenCJ) [reply client/src/pages/Home.js:21]: Which wrapper? the useEffect?

File: client/src/pages/Home.js (hunk 6 above)
Comment (LebkuchenCJ) [reply client/src/pages/Home.js:34]: Otherwise my data will not be visible on each post

File: client/src/components/EmptyListScreen.js (hunk 3 above)
Comment (jamarob) [reply client/src/components/EmptyListScreen.js:17]: Just my preference ;)

File: client/src/pages/Home.js (hunk 7 above)
Comment (LebkuchenCJ) [reply client/src/pages/Home.js:47]: Good point, but I will then have a slight moment of where my empty list container is visible

File: client/src/pages/Home.js (hunk 8 above)
Comment (LebkuchenCJ) [reply client/src/pages/Home.js:50]: For now, I will stick to my method. But I will keep yours in mind. I think I tried it before without success

File: client/src/pages/Home.js (hunk 9 above)
Comment (LebkuchenCJ) [reply client/src/pages/Home.js]: 👍

File: client/src/pages/Home.js (hunk 10 above)
Comment (LebkuchenCJ) [reply client/src/pages/Home.js:67]: That's because it is a form ;)

File: client/src/pages/List.js (hunk 11 above)
Comment (LebkuchenCJ) [reply client/src/pages/List.js:92]: see comment above

**Pull request merged**